compiler/resolution/lifetime.cpp

```diff
@@ -2604,18 +2620,66 @@ bool EmitLifetimeErrorsVisitor::enterCallExpr(CallExpr* call) {
 
             // For the purposes of this check, use scope lifetime
             // if above lifetime is unknown.
-            if (a1lp.borrowed.unknown)
+            if (a1lp.borrowed.unknown) {
               a1lp.borrowed = scopeLifetimeForSymbol(actual1sym);
-            if (a2lp.borrowed.unknown)
+              if (debugging) {
+                printf("Actual %s [%d] lifetime unknown, using scoped "
+                       "lifetime\n",
+                       actual1sym->name, actual1sym->id);
+                printf("%s", "Scoped lifetime is ");
+                printLifetime(a1lp.borrowed);
+                printf("\n");
+              }
+            }
+
+            if (a2lp.borrowed.unknown) {
               a2lp.borrowed = scopeLifetimeForSymbol(actual2sym);
+              if (debugging) {
+                printf("Actual %s [%d] lifetime unknown, using scoped "
+                       "lifetime\n",
+                       actual2sym->name, actual2sym->id);
+                printf("%s", "Scoped lifetime is ");
+                printLifetime(a2lp.borrowed);
+                printf("\n");
+              }
+            }
+
+            // Delay computing to emit debugging information.
+            bool bothFormalsContainBorrowedOrAreBorrowed = false;       
+         
+            if (isOrRefersBorrowedClass(formal1->getValType())) {
+              bothFormalsContainBorrowedOrAreBorrowed = true;
```

Comment:
Nothing's going to set this to `false` when `formal2` isn't borrowed.

Reply:
Good point!